### ISSUE TITLE
Unreadable arguments and "debugging optimized function" warning

### ISSUE THREAD

**Issue by antoineco** [derekparker/delve#1368]
**Environment**

dlv 1.1.0 (`master` @ 4e4ed029)
go 1.11.1
os darwin/amd64

**What did you do?**

With the following test code:

https://gist.github.com/antoineco/262c7dc7860ddd25ef5cd124d2a8ceb3

I ran:

```
$ dlv test ./
Type 'help' for list of commands.
```
```gdb
(dlv) b k8s.io/client-go/testing.(*tracker).add
Breakpoint 1 set at 0x19ec06b for issue/vendor/k8s.io/client-go/testing.(*tracker).add() ./vendor/k8s.io/client-go/testing/fixture.go:309
```
```gdb
(dlv) c
> issue/vendor/k8s.io/client-go/testing.(*tracker).add() ./vendor/k8s.io/client-go/testing/fixture.go:309 (PC: 0x19ec06b)
Warning: debugging optimized function
   304:                 }
   305:         }
   306:         return watches
   307: }
   308:
=> 309: func (t *tracker) add(gvr schema.GroupVersionResource, obj runtime.Object, ns string, replaceExisting bool) error {
   310:         t.lock.Lock()
   311:         defer t.lock.Unlock()
   312:
   313:         gr := gvr.GroupResource()
   314:
```
```
(dlv) args
```

**What did you expect to see?**

The `args` command prints the value of the function arguments.

**What did you see instead?**

`Warning: debugging optimized function` is printed and function arguments are unreadable.

```gdb
(dlv) args
gvr = (unreadable empty OP stack)
ns = (unreadable could not find loclist entry at 0x4bed3 for address 0x1e3050b)
obj = (unreadable could not find loclist entry at 0x4bf0f for address 0x1e3050b)
replaceExisting = (unreadable could not find loclist entry at 0x4bf4a for address 0x1e3050b)
t = (unreadable could not find loclist entry at 0x4bf7f for address 0x1e3050b)
~r4 = (unreadable empty OP stack)
```

**Important note**

⚠️ Here comes the interesting part.

If one un-comments the commented out lines inside `issue.go`, the warning will **not** appear, and  Delve will *consistently* print function arguments:

```gdb
(dlv) args
gvr = issue/vendor/k8s.io/apimachinery/pkg/runtime/schema.GroupVersionResource {Group: "", Version: "v1", Resource: "events"}
ns = ""
obj = issue/vendor/k8s.io/apimachinery/pkg/runtime.Object(*issue/vendor/k8s.io/api/core/v1.Event) 0xc0000f8dd8
replaceExisting = false
t = (*issue/vendor/k8s.io/client-go/testing.tracker)(0xc0002c5400)
~r4 = (unreadable invalid interface type: could not find str field)
```

**Comment by aarzilli** [comment]
Duplicate of https://github.com/golang/go/issues/27681

**Comment by barbu110** [comment]
it still happens to me after golang/go#27681 has been closed

**Comment by j0hnwhyte** [comment]
Same, using the integrated debugger of Goland. The binary is built without optimization, and not all variables are readable.

Using the build flags (go 1.12):
`go build -o=/... -gcflags="all=-N -l" ...`

Error:
`unreadable: could not find loclist entry at 0x... for address 0x...`

**Comment by zhangguanzhang** [comment]
```
 -gcflags="all=-N -l" 
```

**Comment by asg0451** [comment]
> ```
>  -gcflags="all=-N -l" 
> ```

this is amazing. it should be prominently displayed in the docs, as delve can be borderline unusable without it.

**Comment by ArturMarekNowak** [comment]
Happened to me just right now. I migrated from go 1.23 to go 1.25. I updated GoLand to the version 2024.2.5 but it did not update the delve. Delve stayed at 1.23. So I run:

`go install github.com/go-delve/delve/cmd/dlv@latest`

and replaced the dlv.exe by copying from 

`%userprofile%\go\bin`

to 

`C:\Program Files\JetBrains\GoLand 2023.3.5\plugins\go-plugin\lib\dlv\windows`

After that it debugger in GoLand worked as expected. Delve is at 1.25.2
